tests/test_rds.py

```diff
@@ -1254,6 +1254,43 @@ def test_rds_add_security_group(self):
         self.assertEqual(len(clean_resources[0]["VpcSecurityGroups"]), 2)
         self.assertEqual(len(clean_resources), 4)
 
+    def test_rds_filter_by_vpcid(self):
+        #
+        # Test conditions:
+        # Purpose of test is only to validate checking vpc ID with DBSubnetGroup.VpcId
+        # Uses the add_security_group data--should match 2 DB instances
+        # Checks that the expected VPC is present
+
+        session_factory = self.replay_flight_data("test_rds_filter_by_vpcid")
+        p = self.load_policy(
+            {
+                "name": "filter-by-vpcid",
+                "resource": "rds",
+                "filters": [
+                    {"type": "security-group", "key": "GroupName", "value": "default"},
+                    {
+                        "type": "value",
+                        "key": "DBInstanceIdentifier",
+                        "value": "test-sg-fail.*",
+                        "op": "regex",
+                    },
+                    {
+                        "type": "value",
+                        "key": "DBSubnetGroup.VpcId",
```

Comment:
ideally the test would actually use the filter being fixed...

Reply:
oh.. maybe I missing something the pr is changing the vpc filter related ids expression.. but your referencing that its fixing the modify-security-groups which doesn't use that.. ?

Reply:
I'm filtering by the expression, which probably isn't the correct way to unit test this, but it looks like `resolve_group_names` in network.py is not grabbing the vpc_id properly. It's constructing the vpc_id with the `vpc_expr` from the validate function also in network.py. This line (87 for me) shows where it uses `RelatedIdsExpression`:

`self.vpc_expr = jmespath.compile(vpc_filter.RelatedIdsExpression)`

And here is the full stack trace generated when I try to run the policy:

```
019-07-28 22:43:27,070 - custodian.resources.rds - DEBUG - Filtered from 2 to 1 rds
2019-07-28 22:43:27,070 - custodian.policy - INFO - policy: rds-sg-add resource:rds region:us-east-1 count:1 time:0.35
2019-07-28 22:43:27,480 - custodian.output - ERROR - Error while executing policy
Traceback (most recent call last):
  File "/src/c7n/policy.py", line 264, in run
    results = a.process(resources)
  File "/src/c7n/resources/rds.py", line 1379, in process
    rds_instances)
  File "/src/c7n/actions/network.py", line 220, in get_groups
    r, self._get_array('add'), resolved_groups)
  File "/src/c7n/actions/network.py", line 167, in resolve_group_names
    resource_id=r[self.manager.resource_type.id]))
c7n.exceptions.PolicyExecutionError: policy:rds-sg-add non vpc attached resource used with modify-security-group: sample-db-name
```

I'm certain this change in the expression resolves it, but I'm not sure how to better unit test it other than just matching via filter, since it's many layers deep. Thoughts?

Reply:
i think a simple unit test around the vpc filter should suffice since thats what your changing.

Reply:
So I think what I'm understanding now is the `modify-security-groups` action works fine if a different filter is used, but the action itself fails if the `DBSubnetGroup.VpcId` filter is used, because the filter passes a bad list of resources' VPC IDs to the action.

Reply:
the modify security group uses the vpc filter, if you test the vpc filter you'll have tested your change and fix.

Reply:
Isn't my current test already doing this? It's using the "value" filter but it's constructing the expression the same way the "vpc" filter ultimately does, using the jmespath compile from `DBSubnetGroup.VpcId`. In any case, what would be a valid key for the vpc filter? I'm trying `VpcId` for the key and the id for the value but no dice.

Reply:
I changed the filter to vpc, even though the value filter uses the same thing eventually. now it's failing for another reason, not sure what the CacheBeta check does.

Reply:
looks like it's passing now...

Reply:
there was some issue with the azure pipelines cache setup, it looks resolved now.

Reply:
> there's alot of extraneous test data here, please leave off the on the security group filter, the intent is to just test the vpc filter. if you need an example i just did one to fix eni vpc filter in #4463

how's that look?

Reply:
looks good, thanks